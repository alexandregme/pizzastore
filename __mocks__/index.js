export const TOPPINGS = [{
  name:"topping 1",
  price:1,
  isChecked:true
},{
  name:"topping 2",
  price:2,
  isChecked:false
},{
  name:"topping 3",
  price:3,
  isChecked:false
},{
  name:"topping 4",
  price:4,
  isChecked:false
},{
  name:"topping 5",
  price:5,
  isChecked:false
},{
  name:"topping 6",
  price:6,
  isChecked:false
},{
  name:"topping 7",
  price:7,
  isChecked:false
},{
  name:"topping 8",
  price:8,
  isChecked:false
}];

export const DEFAULT_TOPPING = {
  name:"default topping",
  price:0,
  isChecked:false
};

export const CUSTOM_TOPPING = {
  name:"custom topping",
  price:1,
  isChecked:true
};

export const DEFAULT_PIZZA = {
  "name": "default pizza",
  "maxToppings": 0,
  "basePrice": 0,
  "totalPrice": 0,
  toppings:[]
};

export const CUSTOM_PIZZA = {
  "name": "custom pizza",
  "maxToppings": 1,
  "basePrice": 1,
  "totalPrice": 1,
  toppings:[{
    name:"topping 1",
    price:1,
    isChecked:true
  }]
};

export const PIZZAS = [{
  "name": "small",
  "maxToppings": 3,
  "basePrice": 9.89,
  toppings: TOPPINGS
},{
  "name": "medium",
  "maxToppings": 5,
  "basePrice": 10.89,
  toppings: TOPPINGS
}, {
  "name": "large",
  "maxToppings": null,
  "basePrice": 13.49,
  toppings: TOPPINGS
}];
