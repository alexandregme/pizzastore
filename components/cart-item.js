import React from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { removeFromCart } from "../actions";
import Button from "./button";
import { DEFAULT_PIZZA } from "../__mocks__";
import PizzaSVG from '../svgs/pizza.svg';

export const CartItem = ({cartItem = DEFAULT_PIZZA, handleRemoveFromCart = ()=>{}}) =>(
  <React.Fragment>
    <div className="card bg-light text-center col-md-12">
      <div className="card-header col-md-2">
        <PizzaSVG className="pizzaSVG" />
      </div>
      <div className="card-body col-md-8">
        <div className="col-md-6 info">
          <h5 id="pizzaName" className="card-title">Pizza size - {cartItem.name}</h5>
          <p id="pizzaMaxToppings" className="card-text">Max Toppings - {cartItem.maxToppings ? cartItem.maxToppings : 'unlimited toppings!'}</p>
          <p id="pizzaBasePrice" className="card-text">Base price - ${cartItem.basePrice}</p>

        </div>
        <div className="col-md-6 info">
          <ul>
            <li><h5 className="card-title">Toppings</h5></li>
            {
              cartItem.toppings
                .filter(topping => topping.defaultSelected)
                .map((topping, id) => <li key={id}>{topping.topping.name} - ${topping.topping.price}</li>)
            }
          </ul>
        </div>
      </div>
      <div className="card-footer col-md-2">
        <p id="pizzaTotalPrice" className="card-text">Total price - ${cartItem.totalPrice}</p>
        <Button
          label={'remove pizza'}
          className="btn btn-danger"
          handleClick={handleRemoveFromCart}
        />
      </div>
    </div>
  </React.Fragment>
);

CartItem.propTypes = {
  cartItem: PropTypes.object,
  handleRemoveFromCart: PropTypes.func
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  handleRemoveFromCart() { dispatch(removeFromCart(ownProps.cartItem.id)); }
});

export default connect(
  null,
  mapDispatchToProps
)(CartItem);
