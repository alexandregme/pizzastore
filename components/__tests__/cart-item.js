import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { CartItem } from '../cart-item';
import Button from "../button";
import {CUSTOM_PIZZA} from "../../__mocks__";

describe("Cart Item", ()=>{
  let mountedCartItem;

  beforeEach(()=>{
    mountedCartItem = shallow(<CartItem />);
  });

  it('renders without crashing', () => {
    shallow(<CartItem />);
    expect(mountedCartItem).toMatchSnapshot();
  });

  it('expect to have default value for cart item', () => {
    let cartItem = mountedCartItem.find('p');
    expect(cartItem).toHaveLength(1);
    expect(cartItem.text()).toEqual('cart item');
  });

  it('expect to have Button component', () => {
    expect(mountedCartItem.find(Button)).toHaveLength(1);
  });
});

describe("Cart Custom component props", ()=>{
  let mountedCartItem;
  let props = {cartItem: CUSTOM_PIZZA};

  beforeEach(()=>{
    mountedCartItem = shallow(<CartItem {...props} />);
  });

  it('expect to have default value for pizza name', () => {
    let pizzaNameValue = mountedPizza.find('p.pizzaName');
    expect(pizzaNameValue).toHaveLength(1);
    expect(pizzaNameValue.text()).toEqual('custom pizza');
  });

  it('expect to have default value for pizza max toppings', () => {
    let pizzaMaxToppingValue = mountedPizza.find('p.pizzaMaxToppings');
    expect(pizzaMaxToppingValue).toHaveLength(1);
    expect(pizzaMaxToppingValue.text()).toEqual("1");
  });

  it('expect to have default value for base price', () => {
    let pizzaBasePriceValue = mountedPizza.find('p.pizzaBasePrice');
    expect(pizzaBasePriceValue).toHaveLength(1);
    expect(pizzaBasePriceValue.text()).toEqual("1");
  });
});
