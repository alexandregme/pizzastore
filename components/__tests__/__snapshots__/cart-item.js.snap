// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Item renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <p>
          cart item
        </p>,
        <Button
          handleClick={[Function]}
          label="remove item undefined"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "cart item",
        },
        "ref": null,
        "rendered": "cart item",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": [Function],
          "label": "remove item undefined",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p>
            cart item
          </p>,
          <Button
            handleClick={[Function]}
            label="remove item undefined"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "cart item",
          },
          "ref": null,
          "rendered": "cart item",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [Function],
            "label": "remove item undefined",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
